homeassistant/components/elmax/cover.py

```diff
@@ -0,0 +1,137 @@
+"""Elmax cover platform."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from elmax_api.model.command import CoverCommand
+from elmax_api.model.cover_status import CoverStatus
+from elmax_api.model.panel import PanelStatus
+
+from homeassistant.components.cover import CoverEntity, CoverEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ElmaxCoordinator
+from .common import ElmaxEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Elmax cover platform."""
+    coordinator: ElmaxCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    known_devices = set()
+
+    def _discover_new_devices():
+        panel_status: PanelStatus = coordinator.data
+        # In case the panel is offline, its status will be None. In that case, simply do nothing
+        if panel_status is None:
+            return
```

Comment:
```suggestion
        if (panel_status := coordinator.data) is None:
            return # In case the panel is offline, its status will be None. In that case, simply do nothing
```

Reply:
Done in [3be9456](https://github.com/home-assistant/core/pull/79409/commits/3be9456aaadb07f0b007e19ed839c01f7a473271)